components/button/floating-buttons.js

```diff
@@ -0,0 +1,260 @@
+			 * (ex. phones).
+			 */
+			alwaysFloat: {
+				type: Boolean
+			},
+
+			/**
+			 * Minimum height of view-port in order for buttons to float.
+			 */
+			minHeight: {
+				type: String
+			},
+
+			_viewportIsAtLeastMinHeight: {
+				reflect: true,
+				type: Boolean
+			},
+
+			_containerClasses :{
+				type: Object
+			},
+
+			_innerContainerStyle: {
+				type: Object
+			},
+
+			_buttonSpacerStyle: {
+				type: Object
+			}
+		};
+	}
+
+	static get styles() {
+		return [buttonStyles,
+			css`
+			:host {
+				box-sizing: border-box;
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-floating-buttons-container {
+				border-top: 1px solid transparent;
+				display: block;
+				margin: 0 auto;
+				width: 100%;
+			}
+			.d2l-floating-buttons-container.d2l-floating-buttons-floating {
+				animation: d2l-floating-buttons-animation 500ms ease-out;
+				-webkit-animation: d2l-floating-buttons-animation 500ms ease-out;
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+		this._containerClasses = {
+			'd2l-floating-buttons-container': true,
```

Comment:
Since this being true does not seem to be modified, it might be more straightforward to have a variable for `d2l-floating-buttons-floating` instead. e.g. in render have,
```
const containerClasses = {
   'd2l-floating-buttons-container': true,
   'd2l-floating-buttons-floating': this.__floatingButtonsFloating
};
```
and then set `this.__floatingButtonsFloating` as needed

Reply:
`this. _containerClasses` could move into `render` and become `const containerClasses` since it would no longer need to be a property (since you're just modifying `this. _floatingButtonsFloating`